File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        ArrayList<Integer> list = new ArrayList<>();
        int[] p1 = {1,2,3,4,5};
        int[] p2 = {2,1,2,3,2,4,2,5};
        int[] p3 = {3,3,1,1,2,2,4,4,5,5};
        int[] counts = {0, 0, 0};
        for(int i = 0; i < answers.length; i++) {
            if(p1[i%p1.length] == answers[i]) {
                counts[0]++;
            }
            if(p2[i%p2.length] == answers[i]) {
                counts[1]++;
            }
            if(p3[i%p3.length] == answers[i]) {
                counts[2]++;
            }
        }
        int max = 0;
        for(int i = 0; i < 3; i++) {
            max = Math.max(max,counts[i]);
        }
        for(int i = 0; i < 3; i++) {
            if(max == counts[i]) {
                list.add(i+1);
            }
        }
        int[] answer = new int[list.size()];
        for(int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        Arrays.sort(answer);
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    var answer = [];
    const person = [[1,2,3,4,5],[2,1,2,3,2,4,2,5],[3,3,1,1,2,2,4,4,5,5]]
    const obj = {
        1:0,
        2:0,
        3:0
    }
    for (let i = 0; i < answers.length; i++) {
        const answer = answers[i];
        for (let j = 0; j < person.length; j++) {
            if (person[j][i % person[j].length] === answer) obj[j+1]++;
        }
        
    }
    const max = Math.max(...Object.values(obj));
    
    
    for (const key in obj) {
        if (max === obj[key]) answer.push(Number(key));
    }
    answer.sort((a,b)=>a-b);
    return answer;
}